src/style/color.rs

```diff
@@ -153,6 +153,43 @@ impl serde::Serialize for Color {
 
 #[cfg(feature = "serde")]
 impl<'de> serde::Deserialize<'de> for Color {
+    /// This is used to deserialize a value into Color via serde.
+    ///
+    /// This implementation uses the `FromStr` trait to deserialize strings, so named colours, RGB,
+    /// and indexed values are able to be deserialized. In addition, values that were produced by
+    /// the the older serialization implementation of Color are also able to be deserialized.
+    ///
+    /// See the [`Color`] documentation for more information on color names.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use ratatui::prelude::*;
+    ///
+    /// #[derive(serde::Deserialize)]
+    /// struct Theme {
+    ///     color: Color,
+    /// }
+    ///
+    /// let theme: Theme = serde_json::from_str::<Theme>("{\"color\": \"bright-white\"}").unwrap();
+    /// assert_eq!(theme.color, Color::White);
+    ///
+    /// let theme: Theme = serde_json::from_str::<Theme>("{\"color\": \"#00FF00\"}").unwrap();
+    /// assert_eq!(theme.color, Color::Rgb(0, 255, 0));
+    ///
+    /// let theme: Theme = serde_json::from_str::<Theme>("{\"color\": \"42\"}").unwrap();
+    /// assert_eq!(theme.color, Color::Indexed(42));
+    ///
+    /// let theme: Result<Theme, _> = serde_json::from_str::<Theme>("{\"color\": \"invalid\"}");
+    /// assert!(theme.is_err());
```

Comment:
I'd prefer assert_eq!(theme, Err(...)) here if that makes sense.

Reply:
Probably annoying to achieve as the error is some serde_json error

Reply:
Yea the best I can come up with is 
```rust
let theme: Result<Theme, _> = serde_json::from_str(r#"{"color": "invalid"}"#);
assert!(theme.is_err());
if let Err(err) = theme {
    assert_eq!(
        err.to_string(),
        "Failed to parse Colors at line 1 column 20"
    )
}
```
Is this fine, should we leave it, or is there a better way to implement this?

Reply:
Perhaps:
```rust
let err = serde_json::from_str::<Theme>(r#"{"color": "invalid"}"#).unwrap_err();
assert!(err.is_data());
assert_eq!(err.to_string(), "Failed to parse Colors");
```